tasks/move-granules/index.js

```diff
@@ -144,13 +187,12 @@ async function updateGranuleMetadata(granulesObject, collection, cmrFiles, bucke
  */
 async function moveFileRequest(
   file,
-  sourceBucket,
   duplicateHandling,
   markDuplicates = true,
   s3MultipartChunksizeMb
 ) {
   const source = {
-    Bucket: sourceBucket,
+    Bucket: file.sourceBucket,
```

Comment:
Is there any use case where this bites us? I suppose it would need to be a situation where the file has a bucket that isn't what the task wants. So if the file has `sourceBucket1` but the task is passed `sourceBucket2` as almost an override...

I don't know if that's a real thing. Just that move-granules is used quite a lot in ingest workflows so wanna be sure we're covering all our bases.

Reply:
yea, I spent a lot of time thinking about this. it's possible that we want to allow config.bucket as an override, but that has the problem that in *most* cases a granule that is already where it belongs will have files in *multiple* buckets. the case where there is "one master bucket" that all files are stored in is a special case for moving from staging to final location.
but in that case it looks like the granule should have its files correctly written with the staging bucket configured as the file's bucket attribute. maybe I need to follow up to ensure that that's true but that's why it's listed as a breaking change in the changeling